crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -908,6 +909,19 @@ impl<'db> Bindings<'db> {
                             }
                         }
 
+                        Some(KnownClass::TypeAliasType) => {
+                            if let [Some(name), Some(value), ..] = overload.parameter_types() {
+                                // TODO: emit a diagnostic if the name is not a string literal
+                                if let Some(name) = name.into_string_literal() {
+                                    overload.set_return_type(Type::KnownInstance(
+                                        KnownInstanceType::TypeAliasType(TypeAliasType::Bare(
+                                            BareTypeAliasType::new(db, name.value(db), value),
+                                        )),
+                                    ));
+                                }
+                            }
+                        }
```

Comment:
If you move this construction logic into `infer.rs`, like we do for legacy typevars, you'll be able to get the `Definition` of the assignment statement, at least for the case where a call to the `TypeAliasType` constructor is being immediately assigned to a variable. That will let you return a non-`None` result for `definition` up above for bare type aliases too.

https://github.com/astral-sh/ruff/blob/dd04ca7f5885b7bc96303dbb5ce7635a039d2f93/crates/ty_python_semantic/src/types/infer.rs#L5197-L5199

(Moving the construction logic also made it easier to emit diagnostics for legacy typevar construction, which might help you discharge the TODO comment)